Add tests, fix max file size, and update dependencies

nopjmp opened the pull request:
```
Generating cover results ...

Percentage | Module
-----------|--------------------------
     0.00% | Harbor.Disk
     4.35% | Harbor.Plug
    23.08% | Harbor.Maid
   100.00% | Harbor.Application
   100.00% | Harbor.Did
   100.00% | Harbor.Pds
-----------|--------------------------
    43.18% | Total

Coverage test failed, threshold not met:

    Coverage:   43.18%
    Threshold:  90.00%
```

nopjmp committed:
chore: add mock HTTPoison responses and new test fixtures for PDS integration

nopjmp committed:
fix max_file_size check

nopjmp committed:
fix: use configurable cache folder for file sweeping

nopjmp committed:
chore: mix format

nopjmp committed:
fix: update Elixir version in Dockerfile to 1.18.3

nopjmp committed:
fix: update dependencies in mix.lock to latest versions

purifetchi commented:
looks fantastic!

could we get some CI going on here as well? to force test checking on MRs pushing to main

nopjmp commented:
@purifetchi do you want the pipeline to show the test coverage as a comment? I will get tests going at least.

nopjmp committed:
feat: add GitHub Actions workflow

nopjmp committed:
update checkout to v4 and use commit hash for setup-beam versioning

This avoids a supply chain attack where someone force pushes an update to a repo and can extract secrets.

purifetchi commented:
> @purifetchi do you want the pipeline to show the test coverage as a comment? I will get tests going at least.

sure!

nopjmp committed:
feat: enhance GitHub Actions workflow with coverage feedback

nopjmp commented:
I do not believe the pipeline will run until it's in the repo itself due to CI/CD security, but I believe this should work. Future versions, or another workflow, can publish a docker container to GitHub's container storage.

purifetchi commented:
fantastic work, thank you so much!
